core/src/main/java/com/oath/oak/Handle.java

```diff
@@ -39,16 +39,18 @@ boolean isDeleted() {
         return deleted.get();
     }
 
-    boolean remove(MemoryManager memoryManager) {
+    <V> Result<V> remove(MemoryManager memoryManager, V oldValue, Function<ByteBuffer, V> transformer) {
         writeLock.lock();
-        if (isDeleted()) {
+        V vv = (transformer != null) ? transformer.apply(value) : null;
```

Comment:
Shouldn't the read be only after the deleted bit is checked?
It seems otherwise that you might read someone else's value, which may throw IndexOutOfBoundsException.

Reply:
You're right, will fix